### ISSUE TITLE
[6.2 Sticky] When anchor is below the sticky element on the page, calculation is off by its height

### ISSUE THREAD

**Issue by acarnwath** [zurb/foundation-sites#8396]
When a sticky element (in it's non-sticky state) naturally pushes its anchor down the page, the anchor's position is off by the element's height.

The order is as follows:

1. `sticky` class is added on init
   * This applies `position: absolute;` to the sticky element, which causes the sticky container to be 0 height
   * This causes the content below the sticky to move up, underneath the sticky container
2. position of anchor is calculated
   * The current position of the anchor (if below the sticky element) is off by the height of the element
3. height of the sticky container is calculated and added

By moving the `this.$element.addClass(this.options.stickyClass);` into the `load.zf.sticky` handler in the callback of `_this.setSizes`, it is able to get the proper "natural" position of the anchor.

Subsequent `_this.setSizes` calls act as expected, since the container has had its height set (it is only the initial, without the height and without the sticky element in the flow, that it causes issues).

**Comment by ar-rm** [comment]
I think I might have run into this from *two* different directions.

tl;dr: What you originally ran into might be two *different* bugs, not one.

Check out these issues:
- #8825: [Sticky] Wrong default value for data-check-every at documentation
- #8791: [Sticky] Sticky causing content below to "jump" on load


> When a sticky element (in it's non-sticky state) naturally pushes its anchor down the page, the anchor's position is off by the element's height.

This is how I first ran into this as well. The sticky element was un-sticking too early.

I deduced (partially correctly) that the issue was with the JS not recalculating the height *after* it changes the stuck/unstuck state. This lead me to the `data-check-every` value. Which leads right to #8825. The docs say the default should be set to `50`, but it's actually set to `-1`. And `-1` means no recalculating on scroll. Hence this issue. I set it to something other than `-1` and this is no longer an issue. The element becomes unstuck at the right spot.

But you also mention:
```
sticky class is added on init
    - This applies position: absolute; to the sticky element, which causes the sticky container to be 0 height
This causes the content below the sticky to move up, underneath the sticky container
```

Which I *also* ran into.
It's not related (I don't think) to #8825, but it is related to #8791. The built-in `.sticky` class is *probably* setting `position: absolute` unnecessarily. Which is causing the initial jump you mentioned.

I see that this is closed and a merge has been made. I don't know enough to know if that is the right thing to do. It seems right. But I thought it was worth mentioning what I found, so I'm posting it here just incase and for (possible) completeness.

 I hope this helps.
